### PR TITLE
based on shared_with field of POST model we will manage public, team

### PR DESCRIPTION
**Pull request by yatishTrootech**
```
Team - Only the users in the Creator Department will be able to see this Appreciation
Public - everyone in the org can see.
```

### CODE REVIEW

File: feeds/views.py [hunk 1]
```diff
@@ -839,6 +839,18 @@ class UserFeedViewSet(viewsets.ModelViewSet):
...
+    def get_filtered_feeds_according_to_shared_with(feeds, user):
+        """
+        Returns filtered queryset (same dept posts will be returned if post is shared with departments)
+        (All posts will be returned if shared with within organization)
+        (hide posts which has shared with is admin only)
+        params: posts: QuerySet[Post]
+        params: user: CustomUser
+        """
+        feeds = feeds.exclude(id__in=posts_not_shared_with_self_department(feeds, user).values_list("id", flat=True))
+        return feeds.exclude(shared_with=SHARED_WITH.ADMIN_ONLY)
```
Comment (prashant-rewardz) [inline feeds/views.py:852]: if user is admin, then no need to exclude that feeds.

Comment (prashant-rewardz) [inline feeds/views.py:852]: if user is admin, then no need to exclude that feeds.
admin user can see ADMIN_ONLY status feeds.

Comment (yatishTrootech) [reply feeds/views.py]: Hey @prashant-rewardz I have fixed the admin related things can you please review PR again

Comment (yatishTrootech) [reply feeds/views.py]: implemented private means:
creator+cc+admin (is_staff=True)

@sudhanshut1981 @Surajrewardz @qanaveen1111

**Review by prashant-rewardz — APPROVED** [review]

**Pull request merged**